### ISSUE TITLE
Seamless Audiotrack change, current player time position

### ISSUE THREAD

**Issue by pedrosilvabos** [google/shaka-player#3295]
**Have you read the [Tutorials](http://shaka-player-demo.appspot.com/docs/api/tutorial-welcome.html)?**
YES

**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
YES

**What version of Shaka Player are you using?**
3.0.10

**Please ask your question**
Hello, I am able to change the audio language for my DASH + Widevine, using the preferredAudioLanguage in the player configurations. However, as explained in the tutorial, it requires the player to be reloaded.
Is there a way to change the audio language without reloading the player? beeing it a seamless transition?

I am able to predetermine a start position on a VOD through an argument on the load that specifies the seconds the video starts. Throught the seekRange() i can see the min value (0) and the end time. And I am trying to get the current time where the player is. I have read other issues regarding the same time but I can't figure out how to do this.

Thank you

**Comment by michellezhuogg** [comment]
Hello @pedrosilvabos ,

1. If you want to select an audio language after the playback starts, you can use `selectAudioLanguage()`.  See the APIs [here](https://shaka-player-demo.appspot.com/docs/api/shaka.Player.html). 
    If you want to set the 'preferredAudioLanguage' configuration, the player has to load the manifest again to apply that.
2. You can get the current time via `video.currentTime`. See the MDN doc here:
    https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/currentTime 

Let us know if that helps :)

**Comment by pedrosilvabos** [comment]
Hello @pedrosilvabos 
That was perfect! Thank you so much.
Just one last question, and this might be a little off since I understand you develop shaka player and not any specific integration with any platform.
I am developing for WebOS and Tizen, and I am trying to use player.trickPlay() to do fast forwarding or rewind. 
In WebOs negative values on trickplay, in fact work it rewinds, positive values have no effect, however,  it works on Chrome browser.
I used the getplayrate method to debug and I can see that the values are set. However an increase on the trickplay produces no effect, player.trickPlay(-2) works player.trickPlay(2) does not work on Webos. Do you think this could be a "webos" limitation? 
Perhaps its silly, but I am going to try and mitigate this having a function that just cycles the video current time. Do you know of any other way to make this work ?  

Thank you so much for your support and the quick answer!

**Comment by joeyparrish** [comment]
@pedrosilvabos, for fast-forward, we set playbackRate on the video element.  But no browser supports negative playbackRate, so we simulate it with repeated seeking on an interval to rewind.

We don't have a WebOS device, so we were unaware that it didn't support playbackRate to fast-forward.  If this is the case, you could add a check for WebOS and use the same method for fast-forward that we do for rewind.

The code is a little (unnecessarily) abstracted at the moment, but it sounds like you would want to modify `lib/media/play_rate_controller.js`.  In `apply_()`, it has a check for `(rate >= 0)`.  Here's my suggestion based on what you've told us:

```js
    let canSetRate = false;
    if (shaka.util.Platform.isWebOS()) {
      // WebOS doesn't support playbackRate, so only set rates of 1 or 0 here.
      canSetRate = rate == 1 || rate == 0;
    } else {
      // For other platforms, any non-negative rate is acceptable.
      canSetRate = rate >= 0;
    }
    // If canSetRate is false, a timer will be used to seek to simulate the rate.

    if (canSetRate) {
```

You would also need to `goog.require('shaka.util.Platform')` at the top of the file.

I can't test this fix, since we don't have WebOS in our lab, but try that out and let us know how it works.  If it works, you could send us a PR and we would include it in our next release.

**Comment by pedrosilvabos** [comment]
Thanks @joeyparrish 
Currently I ended up going a different route (probably less performant) that captures the arrow key the user is pressing on the remote, pauses the video and just cycles on the timeline, once the user lets go of the button for some milliseconds I just set that time on the shaka.currentTime, hence the video jumps to the specific time.

However, I will check what you said and will come back to you if it works.
Thank you so much.
I have a new question, but I am going to open a new issue for it.
